<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>camera_viewer_form</class>
 <widget class="QWidget" name="camera_viewer_form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1149</width>
    <height>863</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>${DESC}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="5,0">
     <item>
      <layout class="QVBoxLayout" name="left_column_layout" stretch="0,0">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="PyDMLabel" name="PyDMLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string/>
           </property>
           <property name="text">
            <string>Camera:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="camera_description_label">
           <property name="text">
            <string>25idgigeC</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="settings_button">
           <property name="text">
            <string>Settings</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="ImageView" name="image_view" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QFrame" name="controls_frame">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="camera_group_box">
          <property name="title">
           <string>Camera</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="4" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Shape:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <property name="spacing">
                <number>0</number>
               </property>
               <item>
                <widget class="PyDMLabel" name="PyDMLabel_5">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="text">
                  <string>1024</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="channel" stdset="0">
                  <string>haven://${AD}.cam.array_size.array_size_y</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Ã—</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="PyDMLabel" name="PyDMLabel_3">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="text">
                  <string>1224</string>
                 </property>
                 <property name="channel" stdset="0">
                  <string>haven://${AD}.cam.array_size.array_size_x</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Exposure:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Gain:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="PyDMSlider" name="PyDMSlider_2">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="rules" stdset="0">
                  <string>[]</string>
                 </property>
                 <property name="channel" stdset="0">
                  <string>haven://${AD}.cam.gain</string>
                 </property>
                 <property name="userDefinedLimits" stdset="0">
                  <bool>true</bool>
                 </property>
                 <property name="userMinimum" stdset="0">
                  <double>0.000000000000000</double>
                 </property>
                 <property name="userMaximum" stdset="0">
                  <double>20.000000000000000</double>
                 </property>
                 <property name="num_steps" stdset="0">
                  <number>101</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="PyDMPushButton" name="PyDMPushButton_3">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="text">
                  <string>Auto</string>
                 </property>
                 <property name="channel" stdset="0">
                  <string>haven://${AD}.cam.gain_auto</string>
                 </property>
                 <property name="pressValue" stdset="0">
                  <string>1</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="0" column="1">
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="PyDMSlider" name="PyDMSlider">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="rules" stdset="0">
                  <string>[]</string>
                 </property>
                 <property name="channel" stdset="0">
                  <string>haven://${AD}.cam.acquire_time</string>
                 </property>
                 <property name="userDefinedLimits" stdset="0">
                  <bool>true</bool>
                 </property>
                 <property name="userMinimum" stdset="0">
                  <double>0.000000000000000</double>
                 </property>
                 <property name="userMaximum" stdset="0">
                  <double>0.200000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="PyDMPushButton" name="PyDMPushButton_4">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="text">
                  <string>Auto</string>
                 </property>
                 <property name="channel" stdset="0">
                  <string>haven://${AD}.cam.acquire_time_auto</string>
                 </property>
                 <property name="pressValue" stdset="0">
                  <string>1</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="PyDMPushButton" name="PyDMPushButton">
               <property name="toolTip">
                <string/>
               </property>
               <property name="text">
                <string>Start</string>
               </property>
               <property name="rules" stdset="0">
                <string>[{&quot;name&quot;: &quot;enable_acquire_buttons&quot;, &quot;property&quot;: &quot;Enable&quot;, &quot;initial_value&quot;: &quot;False&quot;, &quot;expression&quot;: &quot;ch[0] == 0&quot;, &quot;channels&quot;: [{&quot;channel&quot;: &quot;${PREFIX}cam1:Acquire&quot;, &quot;trigger&quot;: true, &quot;use_enum&quot;: false}]}]</string>
               </property>
               <property name="alarmSensitiveContent" stdset="0">
                <bool>false</bool>
               </property>
               <property name="alarmSensitiveBorder" stdset="0">
                <bool>false</bool>
               </property>
               <property name="channel" stdset="0">
                <string>haven://${AD}.cam.acquire</string>
               </property>
               <property name="passwordProtected" stdset="0">
                <bool>false</bool>
               </property>
               <property name="password" stdset="0">
                <string/>
               </property>
               <property name="protectedPassword" stdset="0">
                <string/>
               </property>
               <property name="showConfirmDialog" stdset="0">
                <bool>false</bool>
               </property>
               <property name="confirmMessage" stdset="0">
                <string>Are you sure you want to proceed?</string>
               </property>
               <property name="pressValue" stdset="0">
                <string>1</string>
               </property>
               <property name="releaseValue" stdset="0">
                <string>None</string>
               </property>
               <property name="relativeChange" stdset="0">
                <bool>false</bool>
               </property>
               <property name="writeWhenRelease" stdset="0">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="PyDMPushButton" name="PyDMPushButton_2">
               <property name="toolTip">
                <string/>
               </property>
               <property name="text">
                <string>Stop</string>
               </property>
               <property name="rules" stdset="0">
                <string>[{&quot;name&quot;: &quot;enable_acquire_buttons&quot;, &quot;property&quot;: &quot;Enable&quot;, &quot;initial_value&quot;: &quot;&quot;, &quot;expression&quot;: &quot;ch[0] == 1&quot;, &quot;channels&quot;: [{&quot;channel&quot;: &quot;${PREFIX}cam1:Acquire&quot;, &quot;trigger&quot;: true, &quot;use_enum&quot;: false}]}]</string>
               </property>
               <property name="channel" stdset="0">
                <string>haven://${AD}.cam.acquire</string>
               </property>
               <property name="pressValue" stdset="0">
                <string>0</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Overlays</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="PyDMTabWidget" name="PyDMTabWidget_2">
             <property name="toolTip">
              <string/>
             </property>
             <property name="whatsThis">
              <string/>
             </property>
             <property name="currentIndex">
              <number>0</number>
             </property>
             <property name="alarmChannels" stdset="0">
              <stringlist>
               <string></string>
               <string></string>
               <string></string>
              </stringlist>
             </property>
             <widget class="QWidget" name="tab_6">
              <attribute name="title">
               <string>Cursor 1</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay_8">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},OV=6</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_overlay.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_7">
              <attribute name="title">
               <string>Cursor 2</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_12">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay_7">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},OV=7</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_overlay.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_8">
              <attribute name="title">
               <string>Datetime</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_4">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay_6">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},OV=8</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_overlay.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>ROIs</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="PyDMTabWidget" name="PyDMTabWidget">
             <property name="toolTip">
              <string/>
             </property>
             <property name="whatsThis">
              <string/>
             </property>
             <property name="tabPosition">
              <enum>QTabWidget::South</enum>
             </property>
             <property name="tabShape">
              <enum>QTabWidget::Rounded</enum>
             </property>
             <property name="currentIndex">
              <number>3</number>
             </property>
             <property name="tabBarAutoHide">
              <bool>false</bool>
             </property>
             <property name="currentTabAlarmChannel" stdset="0">
              <cstring></cstring>
             </property>
             <property name="alarmChannels" stdset="0">
              <stringlist>
               <string></string>
               <string></string>
               <string></string>
               <string></string>
               <string></string>
              </stringlist>
             </property>
             <widget class="QWidget" name="tab">
              <attribute name="title">
               <string>Full</string>
              </attribute>
              <layout class="QHBoxLayout" name="horizontalLayout_11">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay_5">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},R=4</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_roi.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_2">
              <attribute name="title">
               <string>1</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_8">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},R=1</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_roi.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_5">
              <attribute name="title">
               <string>2</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_9">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay_2">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},R=2</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_roi.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_4">
              <attribute name="title">
               <string>3</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_10">
               <item>
                <widget class="PyDMEmbeddedDisplay" name="PyDMEmbeddedDisplay_3">
                 <property name="toolTip">
                  <string/>
                 </property>
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="macros" stdset="0">
                  <string>AD=${AD},R=3</string>
                 </property>
                 <property name="filename" stdset="0">
                  <string>area_detector_roi.ui</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PyDMTabWidget</class>
   <extends>QTabWidget</extends>
   <header>pydm.widgets.tab_bar</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PyDMLabel</class>
   <extends>QLabel</extends>
   <header>pydm.widgets.label</header>
  </customwidget>
  <customwidget>
   <class>PyDMEmbeddedDisplay</class>
   <extends>QFrame</extends>
   <header>pydm.widgets.embedded_display</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PyDMPushButton</class>
   <extends>QPushButton</extends>
   <header>pydm.widgets.pushbutton</header>
  </customwidget>
  <customwidget>
   <class>PyDMSlider</class>
   <extends>QFrame</extends>
   <header>pydm.widgets.slider</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ImageView</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
